cmd/fleet/handleEnroll.go

```diff
@@ -347,6 +346,7 @@ func decodeEnrollRequest(data io.Reader) (*EnrollRequest, error) {
 
 	// Validate
 	switch req.Type {
+	// TODO: Should these be converted to constant? Need to be kept in sync with Kibana?
```

Comment:
@nchaulet @blakerouse Thoughts?

Reply:
@nchaulet Is this even used?

Reply:
No it's not probably no reason to keep it :)

Reply:
Not sure what the paths forward here is. Can the validation part be removed?  I assume the decode we still need.

Reply:
I think we could make the `type` field optional in Kibana (we already do not use it), remove it from the agent, and remove it from fleet server.

Reply:
I did not touch it for now and merged the PR. It sounds removing it might break things in Agent or Fleet.